drivers/wifi/wfx200/wfx200_events.c

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2022 grandcentrix GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME wifi_wfx200_event
+#define LOG_LEVEL CONFIG_WIFI_LOG_LEVEL
+
+#include <sl_wfx.h>
+#undef BIT
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <zephyr.h>
+#include <net/wifi_mgmt.h>
+
+#include <wfx200_internal.h>
+
+void wfx200_event_thread(void *p1, void *p2, void *p3)
+{
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+	struct wfx200_data *context = p1;
+	struct wfx200_queue_event *event;
+
+	k_thread_name_set(NULL, "wfx200_ev");
+
+	while (true) {
+		event = k_queue_get(&context->event_queue, K_FOREVER);
+
+		switch (event->ev) {
+		case WFX200_CONNECT_EVENT:
+			if (context->state == WFX200_STATE_STA_MODE) {
+				net_if_set_link_addr(context->iface, context->sl_context.mac_addr_0.octet,
+						     sizeof(context->sl_context.mac_addr_0.octet),
+						     NET_LINK_ETHERNET);
+				net_eth_carrier_on(context->iface);
+				wifi_mgmt_raise_connect_result_event(context->iface, 0);
+			}
+			if (IS_ENABLED(CONFIG_WIFI_WFX200_SLEEP) &&
+			    !(context->sl_context.state & SL_WFX_AP_INTERFACE_UP)) {
+				/* Enable the power save */
+				sl_wfx_set_power_mode(IS_ENABLED(WIFI_WFX200_PM_MODE_PS) ?
+						      WFM_PM_MODE_PS :
+						      WFM_PM_MODE_DTIM,
+						      IS_ENABLED(WIFI_WFX200_PM_POLL_STRATEGY_UAPSD) ?
+						      WFM_PM_POLL_UAPSD :
+						      WFM_PM_POLL_FAST_PS,
+						      CONFIG_WIFI_WFX200_DTIM_SKIP_INTERVAL
+						      );
+				sl_wfx_enable_device_power_save();
+			}
+			break;
+		case WFX200_CONNECT_FAILED_EVENT:
```

Comment:
When working with WFX200 and some hotspot, sometimes callbacks never happen (e.g. connection request is successfully sent, but no WFX200_CONNECT_EVENT or WFX200_CONNECT_FAILED_EVENT occurs).

For such (and other) it would be meaningful to be able restart the chip. 
This can be done just by calling sl_wfx_init.
It would make sense to provide API for that as well. 
Even though I'm not sure if it should be done here or by extending `net_wifi_mgmt_offload`

Without this remedy, it wasn't possible to reach production quality stability (wfx200 doesn't seem to have internal connection timeout, or it's not always working).

I agree that it should be fixed in WFX200, but possibility to reboot slave chip should also be exposed.